### ISSUE TITLE
Cannot download file greater than 4Gb

### ISSUE THREAD

**Issue by maxivak** [capistrano/sshkit#382]
I am downloading a big file from a server. It stops downloading on 4gb.

```
host = SSHKit::Host.new("#{ssh_user}@#{hostname}")
host.password = 'mypass'

on host do |host|
          download!  "/tmp/1.tar",  "/tmp/local.tar"
end
```

It hangs on 30%.
Size of downloaded file is exactly 4Gb wile the original file size is 11 Gb.

```
ls -lah /tmp

-rwxrwxr-x  1 gex  gex      4,0G  dec 20 13:04 1.tar
```

**Comment by leehambley** [comment]
I think you might have to report this with https://github.com/net-ssh/net-ssh which we build on top of.

I see two potential issues.

- Either we're blocking at 4G on some integer that should be a "bignum"
- We're hitting a timeout at a really suspiscious point.

**Comment by mattbrictson** [comment]
It may also be a net-scp issue: https://github.com/net-ssh/net-scp (that's the gem we use for the actual download implementation).

You could try using `Net::SCP.download!` directly (i.e. without sshkit) and see if you can reproduce the problem there.

**Comment by rimenes** [comment]
@mattbrictson You were right. This was an issue in net-scp but fortunately it was fixed in 5.0.0.rc1: https://github.com/net-ssh/net-scp/issues/39

**Comment by mattbrictson** [comment]
Great! Have you had any trouble using net-scp 5.0.0.rc1 with SSHKit? If not, I think it is safe to close this issue.

**Comment by rimenes** [comment]
Oops. The actual fix was in net-ssh. The fix landed in version 5.0.0.rec1 but they already have released net-ssh 5.0.1.

Answering the question: I didn't have any problem using it.
